flutter-idea/src/io/flutter/vmService/VmServiceWrapper.java

```diff
@@ -473,45 +485,105 @@ public void addBreakpoint(@NotNull final String isolateId,
     addRequest(() -> {
       final int line = position.getLine() + 1;
 
-      final Collection<String> scriptUris = myDebugProcess.getUrisForFile(position.getFile());
+      final String resolvedUri = getResolvedUri(position);
+      final List<String> resolvedUriList = List.of(resolvedUri);
+      System.out.println(resolvedUriList);
+
       final CanonicalBreakpoint canonicalBreakpoint =
         new CanonicalBreakpoint(position.getFile().getName(), position.getFile().getCanonicalPath(), line);
       canonicalBreakpoints.add(canonicalBreakpoint);
       final List<Breakpoint> breakpointResponses = new ArrayList<>();
       final List<RPCError> errorResponses = new ArrayList<>();
 
-      for (String uri : scriptUris) {
-        myVmService.addBreakpointWithScriptUri(isolateId, uri, line, new AddBreakpointWithScriptUriConsumer() {
-          @Override
-          public void received(Breakpoint response) {
-            breakpointResponses.add(response);
-            breakpointNumbersToCanonicalMap.put(response.getBreakpointNumber(), canonicalBreakpoint);
+      myVmService.lookupPackageUris(isolateId, resolvedUriList, new UriListConsumer() {
+        @Override
+        public void received(UriList response) {
+          final List<String> uris = response.getUris();
```

Comment:
need to make sure you aren't disposed by the time this callback is triggered.

Reply:
Check that project is not disposed?